taichi/backends/cuda/cuda_profiler.cpp

```diff
@@ -0,0 +1,150 @@
+#include "taichi/backends/cuda/cuda_profiler.h"
+#include "taichi/backends/cuda/cuda_driver.h"
+#include "taichi/backends/cuda/cuda_context.h"
+
+TLANG_NAMESPACE_BEGIN
+
+CUDAProfiler::CUDAProfiler() {
+  TI_TRACE("CUDAProfiler::CUDAProfiler() ");
+}
+
+CUDAProfiler::~CUDAProfiler() {
+  if (profiler_config_.profiler_type == CUDA_KERNEL_PROFILER_CUPTI) {
+    end_profiling();
+    deinit_cupti();
+  }
+}
+
+bool CUDAProfiler::set_profiling_mode(KernelProfilerMode profiling_mode) {
+  bool enable = profiling_mode == KernelProfilerMode::enable |
```

Comment:
I'd encapsulate this logic into a method on KernelProfilerMode::is_enabled() so that we only need to change one place when adding a new mode.

Reply:
I encapsulated this logic into `is_cuda_profiler()` and simplified init logic. thanks a lot :smile: